src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingScriptOperation.cs

```diff
@@ -246,27 +246,20 @@ private string GetServerNameFromLiveInstance(string connectionString)
         {
             string serverName = null;
             using(SqlConnection connection = new SqlConnection(connectionString))
-            using (SqlCommand cmd = connection.CreateCommand())
-            {
+            { 
                 connection.Open();
 
                 try
                 {
-                    cmd.CommandText = "select @@servername";
-                    serverName = (string)cmd.ExecuteScalar();
+
+                    ServerConnection serverConnection = new ServerConnection(connection);
+                    serverName = serverConnection.TrueName;
                 }
                 catch (SqlException e)
                 {
-                    //
-                    // Azure SQL Data Warehouse does not support @@servername, so fallback to SERVERPROPERTY.
-                    //
-
                     Logger.Write(
                         LogLevel.Verbose, 
-                        string.Format("Exception running query 'SELECT @@servername' {0}, fallback to SERVERPROPERTY query", e));
-
-                    cmd.CommandText = "select SERVERPROPERTY('ServerName') AS ServerName";
```

Comment:
ServerConnection is using SERVERPROPERTY('ServerName') so not need to try again here. If it fails here it means it already failed in the server connection